ios/Sources/MediaFeature/MediaFeature.swift

```diff
@@ -0,0 +1,114 @@
+import Component
+import ComposableArchitecture
+import Model
+
+public struct MediaState: Equatable {
+    var listState: MediaListState?
+
+    public init() {}
+}
+
+public struct MediaListState: Equatable {
+    var list: MediaList
+    var searchText: String?
+}
```

Comment:
It was the original structure of `MediaState`, but it seems like will be loaded at once.
https://github.com/DroidKaigi/conference-app-2021/blob/b10e59dd853765f047005f03166472470c19c874/model/src/commonMain/kotlin/io/github/droidkaigi/feeder/repository/FeedRepository.kt#L7-L10 
And I planed to implement something like below.
``` swift
struct InitialLoadingState<T: Equatable>: Equatable {
    case initial // This may not be neccessary
    case loading
    case failed(errorTitle: String)
    case completed(T)
}
```
But we don't have the design of the loading and failed state, so I just made it an `Optional`.
Is this ok for you, or you have a better idea, please tell me. 🙏

Reply:
> But we don't have the design of the loading and failed state, so I just made it an Optional.

How about use just each list's empty state and switch showing state with `if-else` ?
I think it's not needed to use `Optional`.


And these `struct` s are so nested I think. I think there will be at most two like these.

Two structs case
```swift
// Judge showing state with `mediaList` Optional state.
public struct MediaState {
  public var searchText: String?
  public var mediaList: MediaList?
}

public struct MediaList {
  public var blogs: [FeedItem]
  public var videos: [FeedItem]
  public var podcasts: [FeedItem]
}
```

Only one struct case
```swift
// Judge showing state with each items empty state. (Default value is empty)
public struct MediaState {
  public var searchText: String?
  public var blogs: [FeedItem]
  public var videos: [FeedItem]
  public var podcasts: [FeedItem]
}
```

Reply:
> Judge showing state with each items empty state. (Default value is empty)

Did you mean showing `ProgressView` when all of them are empty?
It may be fine when there must be at least one `FeedItem`, but how to descript the error state for the initial loading?

Reply:
>  how to descript the error state for the initial loading?

As you mentioned before,  we don't have the design of the loading and failed state, so I think we don't have to think about that.

If we should descript `loading` and `failed` loading state, it will be fine with implementing with `SwitchStore` and `enum State`.

Reply:
Sorry, I didn't notice your replay and continue implementing. 💦
And I finally get your point:
- `enum State` (including `Optional`) if fine
- But `mediaList: MediaList?` is weird, and even when handing the loading state I should use like `Store<MediaState?, MediaAction>`

So please let me fix it by the next PR which also replaces `IfLetStore` with `SwitchStore`.